src/network/grpc/client.rs

```diff
@@ -46,7 +46,8 @@ pub fn run_connect_socket(peer: Peer, state: GlobalState) -> impl Future<Item =
                     state
                         .channels
                         .block_box
-                        .send_to(BlockMsg::AnnouncedBlock(header));
+                        .clone()
```

Comment:
It's enough to make `state` mutable, no need for cloning.

Reply:
ha yes. I've been lazy there. I missed that point